File: workout-planning-project/src/main/java/com/netcracker/model/documents/Exercise.java
package com.netcracker.model.documents;

import com.arangodb.springframework.annotation.Document;
import com.arangodb.springframework.annotation.Field;
import com.arangodb.springframework.annotation.Relations;
import com.fasterxml.jackson.annotation.JsonIgnore;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import org.springframework.data.annotation.Id;

import java.util.List;

import static com.netcracker.model.CollectionsNames.EXERCISE;

@Data
@Document(EXERCISE)
@ApiModel(value = "Describes an exercise")
public class Exercise {
    public static final String[] muscleLoad = {"hips", "biceps", "abs", "chest", "shoulders", "back"};

    @Id
    @ApiModelProperty(hidden = true)
    private String id;

    @Field("measure-list")
    private List<String> measureList;

    private String name;
    private Description description;
    private InfForRecommendation infForRecommendation;

    @Data
    public static class Description {
        String technique;
        String features;
    }

}

File: workout-planning-ng-app/src/app/models/exercise.ts
export class Exercise {
  constructor(public id?: string,
              public name?: string,
              public description?: Description,
              public measureList?: string[],
              public infForRecommendation?: InfForRecommendation
  ) { }
}

export interface Description {
  technique: string;
  features: string;
}

export interface InfForRecommendation {
  complexity: number;
  muscleLoad: Map<string, number>;
}
